frontend2/src/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { AuthStateEnum, useCurrentUser } from "../contexts/CurrentUserContext";
+import { Outlet, useNavigate } from "react-router-dom";
+
+const PrivateRoute: React.FC = () => {
+  const { authState } = useCurrentUser();
+  const navigate = useNavigate();
+
+  if (authState === AuthStateEnum.AUTHENTICATED) {
+    return <Outlet />;
+  } else if (authState === AuthStateEnum.NOT_AUTHENTICATED) {
+    navigate("/login");
+    return null;
+  } else {
+    return <p>loading</p>;
```

Comment:
Should we have a more populated loading screen here (e.g. using `Spinner`)? Not necessary though, we can just go back  and add one later if we need...

Reply:
Yup, i'll change this to a centered `Spinner` in the page